### PR TITLE
Add "Code injection attacks" security consideration

### LINKED ISSUE #1731: Missing specification on rpId validations when calling credentials.get() from a different origin
3rd party javascript running on a page is able to call Webauthn API using the rpId of the 1st party origin context.
This allows attackers with control over any JS embedded in a page or through browser extension to steal a user's assertion and perform an account takeover attack.

This was proven to be exploitable on all browsers/platforms/authenticators that support Webauthn.
We conducted the attack by wrapping the `navigator.credentials.get` function and extending its functionality to replace the original PublicKeyCredentialRequestOptions with a malicious one received from a remote server  

The implementation looks like this:
```
originalNavCredGet = navigator.credentials.get  
navigator.credentials.get = async (challenge) => {
	remoteChallenge = await getRemoteChallenge()    
	assertion = await originalNavCredGet(remoteChallenge)       
	sendAssertion(assrtion)
};
```
This code is executed from a cross-origin JS resource or content script.

A full working POC code will be released in the future.
Chromium bug tracker: https://bugs.chromium.org/p/chromium/issues/detail?id=1297739
Mozilla bug tracker: https://bugzilla.mozilla.org/show_bug.cgi?id=1755579

### PR DESCRIPTION
**Pull request by emlun**
Fixes #1731.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1733.html" title="Last updated on Jun 28, 2022, 2:32 PM UTC (74eb1b6)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1733/0678b0d...74eb1b6.html" title="Last updated on Jun 28, 2022, 2:32 PM UTC (74eb1b6)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -6827,7 +6832,7 @@ to protect the [=attestation object=] from [=man-in-the-middle attacks=].
 
 Under the assumption that a [=registration ceremony=] is completed securely, and that the [=authenticator=] maintains
 confidentiality of the [=credential private key=], subsequent [=authentication ceremonies=] using that [=public key
-credential=] are resistant to [=man-in-the-middle attacks=].
+credential=] are resistant to tampering from [=man-in-the-middle attacks=].
```
Comment (emlun) [inline index.bs:6835]: I'm not sure "from" is the right preposition to use here. Perhaps it should be "by"?

Comment (ve7jtb) [reply index.bs:6835]: by or via are probably better

File: index.bs [hunk 2]
```diff
@@ -6898,6 +6903,46 @@ This issue is similar to the one described in [[#sctn-username-enumeration]]
...
+    For example, user-submitted code hosted on `usercontent.example.org`
+    could exercise any [=credentials=] [=scoped=] to the [=RP ID=] `example.org`.
+    If the [=[RP]=] allows a subdomain {{CollectedClientData/origin}}
+    when [verifying the assertion](#rp-op-verifying-assertion-step-origin),
+    malicious users could use this to launch a [=man-in-the-middle attack=]
+    to obtain valid [=authentication assertions=] and impersonate the victims of the attack.
+
+    Therefore, the [=[RP]=] by default SHOULD NOT allow a subdomain {{CollectedClientData/origin}}
+    when [verifying the assertion](#rp-op-verifying-assertion-step-origin).
+    If the [=[RP]=] needs to allow a subdomain {{CollectedClientData/origin}},
```
Comment (agl) [inline index.bs:6940]: If the RP permits _a_ subdomain to be an acceptable origin when validating the ClientDataJSON then they only need to worry about that subdomain, no? This wording might suggest that allowing any subdomain implies additional requirements on _all_ subdomains?

Comment (emlun) [reply index.bs:6940]: Good point. How about this?

```
If the [=[RP]=] needs to allow a subdomain {{CollectedClientData/origin}},
    then the [=[RP]=] MUST NOT serve untrusted code on any allowed subdomain of an
    [=determines the set of origins on which the public key credential may be exercised|origin=]
    within the [=scope=] of its [=public key credentials=].
```

File: index.bs [hunk 3]
```diff
@@ -6898,6 +6903,46 @@ This issue is similar to the one described in [[#sctn-username-enumeration]]
 and [[#sctn-credential-id-privacy-leak]], and can be mitigated in similar ways.
 
 
+### Code injection attacks ### {#sctn-code-injection}
+
+Any malicious code executing on an
+[=determines the set of origins on which the public key credential may be exercised|origin=]
+within the [=scope=] of a [=[RP]=]' [=public key credentials=]
```
Comment (emlun) [inline index.bs:6910]:
```suggestion
within the [=scope=] of a [=[RP]=]'s [=public key credentials=]
```

File: index.bs (hunk 2 above)
Comment (emlun) [reply index.bs:6940]: Edited to the wording proposed above.

File: index.bs (hunk 1 above)
Comment (emlun) [reply index.bs]: I agree, thanks!

**Review by akshayku — APPROVED** [review]

**Pull request merged**